src/main/java/com/smoothstack/utopiaairlines/controllers/AdminTicket.java

```diff
@@ -0,0 +1,140 @@
+package com.smoothstack.utopiaairlines.controllers;
+
+import com.smoothstack.utopiaairlines.entities.Ticket;
+import com.smoothstack.utopiaairlines.entities.TicketPK;
+import com.smoothstack.utopiaairlines.services.TicketAdminService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.dao.DataIntegrityViolationException;
+import org.springframework.web.bind.annotation.*;
+
+import java.time.LocalDateTime;
+import java.util.List;
+import java.util.Objects;
+
+/**
+ * Controllers for administrative Ticket-related CRUD operations.
+ * @author Joshua Podhola
+ */
+@RestController
+@RequestMapping("/admin/ticket")
+public class AdminTicket {
+    @Autowired
+    private TicketAdminService ticketAdminService;
+
+    @GetMapping("/all")
+    public List<Ticket> getAll() {
+        try {
+            return ticketAdminService.getAll();
+        } catch (Exception e) {
+            e.printStackTrace();
+            return null;
+        }
+    }
+
+    @PostMapping("/create")
+    public String create(@RequestBody Ticket ticket) {
+        try {
+            Ticket ticket1 = ticketAdminService.insert(ticket);
+            if (ticket1 != null) {
+                return "Ticket created.";
+            }
+            return "Ticket already exists.";
+        } catch (Exception e) {
+            e.printStackTrace();
+            return "Error creating ticket.";
+        }
+    }
+
+    @GetMapping("/search/membership_id")
+    public List<Ticket> searchByMembershipNumber(@RequestParam String id) {
```

Comment:
Why are we duplicating traveller functionalities? This all methods are already in Traveller Service

Reply:
Returns something else; in one case you're searching for travellers, the other you're searching for their bookings.

Reply:
It's necessary; you can't expect the end user to know, find, or care about a traveller's database ID, but you can use the membership number to look for it. In this case, you'd use it to look for their tickets, rather than just the traveller itself.